code/modules/research/protolathe.dm

```diff
@@ -202,3 +202,16 @@
 			if(new_item.matter && new_item.matter.len > 0)
 				for(var/i in new_item.matter)
 					new_item.matter[i] = new_item.matter[i] * mat_efficiency
+
+/obj/machinery/r_n_d/protolathe/proc/eject(var/material, var/amount)

```

Comment:
I'd rather not merge boilerplate that could be avoided by moving a single variable definition down one link in the object chain. It's barely a refactor.